libs/langchain-community/src/vectorstores/opensearch.ts

```diff
@@ -256,10 +274,16 @@ export class OpenSearchVectorStore extends VectorStore {
           {
             // map all metadata properties to be keyword
             "metadata.*": {
-              match_mapping_type: "*",
+              match_mapping_type: "string",
               mapping: { type: "keyword" },
             },
           },
+          {
+              "metadata.loc": {
+                  match_mapping_type: "object",
+                  mapping: { type: "object" },
+              },
+          },
```

Comment:
https://github.com/langchain-ai/langchainjs/issues/2036#issuecomment-1645769482 to comply with the fixes.